### ISSUE TITLE
[sklearn#34595] Allow TunedThresholdClassifierCV to assess threshold variability and examine all possible thresholds

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#313]
**Mirrored from [scikit-learn#34595](https://github.com/scikit-learn/scikit-learn/issues/34595)**

Original title: Allow TunedThresholdClassifierCV to assess threshold variability and examine all possible thresholds
Original labels: New Feature, Needs Investigation
Created: 2026-07-29T09:53:40Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the workflow you want to enable

If I understand this correctly TunedThresholdClassifierCV calculates the metric values at the indicated threshold fo each folds, then takes the average metric per threshold, and with that it selects the best threshold. Would it be possible to also determine the std of the threshold? Mostly for analysis. Would be nice to see how this value compares with Elkan's threshold.

The other request: at the moment, threshold takes an array of cut-off points, 100 by default. To examine all possible cut-offs, particularly important with highly imbalanced data, would it be possible to allow "auto" for example, in which case the cut-off values to examine would be unique(probs), like it's done for roc curves?

### Describe your proposed solution

not sure this is the best solution but here it goes:

`_mean_interpolated_score` already builds a full array of per-fold interpolated curves before reducing it with np.mean(..., axis=0).

 The ask would be to also keep `np.std(..., axis=0)` and expose it (e.g. `cv_results_["scores_std` and maybe `best_score_std_` at the chosen threshold) when `store_cv_results=True`.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/110 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
